pkg/minibroker/minibroker.go

```diff
@@ -389,29 +394,48 @@ func (c *Client) provisionSynchronously(instanceID, namespace, serviceID, planID
 
 	// Store any required metadata necessary for bind and deprovision as labels on the resources itself
 	klog.V(3).Infof("minibroker: labeling chart resources with instance %q", instanceID)
-	filterByRelease := metav1.ListOptions{
-		LabelSelector: labels.SelectorFromSet(map[string]string{
-			ReleaseLabel: release.Name,
-		}).String(),
-	}
-	services, err := c.coreClient.CoreV1().Services(namespace).List(context.TODO(), filterByRelease)
+	resources, err := c.helm.ChartClient().ListResources(release)
 	if err != nil {
 		return err
 	}
-	for _, service := range services.Items {
-		err := c.labelService(service, instanceID)
+
+	for _, r := range resources {
+		obj, ok := r.Object.DeepCopyObject().(metav1.Object)
+		if !ok {
+			continue
+		}
+
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{InstanceLabel: instanceID}
+		} else {
+			labels[InstanceLabel] = instanceID
+		}
+		obj.SetLabels(labels)
+
+		data, err := json.Marshal(obj)
 		if err != nil {
 			return err
 		}
-	}
-	secrets, err := c.coreClient.CoreV1().Secrets(namespace).List(context.TODO(), filterByRelease)
-	if err != nil {
-		return err
-	}
-	for _, secret := range secrets.Items {
-		err := c.labelSecret(secret, instanceID)
+
+		var dr dynamic.ResourceInterface
+		if r.Mapping.Scope.Name() == meta.RESTScopeNameNamespace {
+			// Namespaced resources.
+			dr = c.dynamicClient.Resource(r.Mapping.Resource).Namespace(obj.GetNamespace())
+		} else {
+			// Cluster-wide resources.
+			dr = c.dynamicClient.Resource(r.Mapping.Resource)
+		}
```

Comment:
Does something like this work?
```suggestion
		dr := c.dynamicClient.Resource(r.Mapping.Resource)
		if r.Mapping.Scope.Name() == meta.RESTScopeNameNamespace {
			// Namespaced resources.
			dr = dr.Namespace(obj.GetNamespace())
		}
```